docs/root/start/sandboxes/dynamic-configuration-control-plane.rst

```diff
@@ -162,8 +162,10 @@ from ``service1`` to ``service2``:
    :emphasize-lines: 6
    :linenos:
 
-Further down in this file you must also change the configuration snapshot version number from
-``"1"`` to ``"2"`` to ensure Envoy sees the configuration as newer:
+Further down in this file you must also change the configuration snapshot version number. The version
+only has to _change_ it does not need to _increase_ for Envoy to see the configuration as newer. You
+could, for example, change the version from ``"1"`` to ``"0"`` and because it has _changed_ Envoy will
+still use the new (``"0"``) configuration. Here we will change the version from ``"1"`` to ``"2"`` :
```

Comment:
I would like to make it clear that the version doesn't have to "increase" (as this wasn't clear to me when I first tried out this sandbox).

How about in the note we say:

```
    In this example we increment the version, but any change to the version will cause Envoy
    to accept the updated configuration. you could, for example, alternate between ``"1"`` and
    ``"0"``.
```

Reply:
im not sure we need to add that if that is actually the case - my concert was more that "any change to version will update to new config" isnt correct - ie it would revert to the previous config - if it doesnt do that then i think we dont need to worry

Reply:
ill ask around/check other docs and try and clarify how it works

Reply:
TL;DR the version must be different for different resources, and in some cases even if the version is different but the contents is the same, the resource instantiation will be skipped.

In xDS, the version is a property of the resource (i.e., same resource contents should result in the same version, just like hash). Therefore, different resources should have a different version, and Envoy will pick it up and process the new resource.
That said, many xDS servers will send an "incremental" or different version, and Envoy will ingest the resource. Note that there are cases (specific xDS types) in Envoy, that optimize this case to avoid fully ingesting the resource, by comparing the contents of the resource with the resource that is already known to Envoy. In that case Envoy will "skip" these resources.

Reply:
So perhaps saying "there is no notion of version order, it does not need to 'increment' monotonically"?

Reply:
so iiuc incrementing is a good strategy if you want to avoid unexpected outcomes ?

Reply:
> so iiuc incrementing is a good strategy if you want to avoid unexpected outcomes ?

Well... I wouldn't say there aren't side-effects to always change the version. Specifically, from performance perspective, Envoy will need to compare the given resource to the already known resource, and this comparison is not implemented for all xDS types.
I think the best way to think about it is each version_info is a UUID that is dependent on the resource contents.

> So perhaps saying "there is no notion of version order, it does not need to 'increment' monotonically"?

I agree with the quote, but will let others decide if a better wording should be provided.

Reply:
> @adisuissa are there any other docs we can redirect to - i think we want to make some broad indicative statement here and then signpost to a more complex explanation

https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#ack-nack-and-resource-type-instance-version
Specifically:
```
Note that the version for a resource type is not a property of an individual xDS stream but rather a property of the resources themselves. If the stream becomes broken and the client creates a new stream, the client’s initial request on the new stream should indicate the most recent version seen by the client on the previous stream. Servers may decide to optimize by not resending resources that the client had already seen on the previous stream, but only if they know that the client is not subscribing to a new resource that it was not previously subscribed to. For example, it is generally safe for servers to do this optimization for LDS and CDS when the only subscription is a wildcard subscription, and it is safe to do in environments where the clients will always subscribe to exactly the same set of resources.
```